source/renderer/app/components/assets/AssetContent.js

```diff
@@ -0,0 +1,245 @@
+// @flow
+import React, { Component, Fragment as F } from 'react';
+import SVGInline from 'react-svg-inline';
+import classnames from 'classnames';
+import { defineMessages, intlShape } from 'react-intl';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { observer } from 'mobx-react';
+import styles from './AssetContent.scss';
+import { hexToString } from '../../utils/strings';
+
+import copyIcon from '../../assets/images/copy-asset.inline.svg';
+import copyCheckmarkIcon from '../../assets/images/check-w.inline.svg';
+import { ASSET_TOKEN_ID_COPY_FEEDBACK } from '../../config/timingConfig';
+import type { Asset as AssetProps } from '../../api/assets/types';
+
+const messages = defineMessages({
+  fingerprintItem: {
+    id: 'assets.assetToken.item.fingerprint',
+    defaultMessage: '!!!Fingerprint',
+    description: '"fingerprint" item.',
+  },
+  policyIdItem: {
+    id: 'assets.assetToken.item.policyId',
+    defaultMessage: '!!!Policy Id',
+    description: '"policyId" item.',
+  },
+  assetNameItem: {
+    id: 'assets.assetToken.item.assetName',
+    defaultMessage: '!!!Asset name',
+    description: '"assetName" item.',
+  },
+  nameItem: {
+    id: 'assets.assetToken.item.name',
+    defaultMessage: '!!!Name',
+    description: '"name" item.',
+  },
+  tickerItem: {
+    id: 'assets.assetToken.item.ticker',
+    defaultMessage: '!!!Ticker',
+    description: '"ticker" item.',
+  },
+  descriptionItem: {
+    id: 'assets.assetToken.item.description',
+    defaultMessage: '!!!Description',
+    description: '"description" item.',
+  },
+  blank: {
+    id: 'assets.assetToken.item.blank',
+    defaultMessage: '!!!Blank',
+    description: '"Blank" item value.',
+  },
+  settingsCogPopOver: {
+    id: 'assets.assetToken.settings.cogPopOver',
+    defaultMessage:
+      '!!!You can configure the number of decimal places for this native token.',
+    description: 'Asset settings pop over content',
+  },
+  settingsWarningPopOverAvailable: {
+    id: 'assets.warning.available',
+    defaultMessage:
+      '!!!Recommended configuration for decimal places for this native token is available.',
+    description: 'Asset settings recommended pop over content',
+  },
+  settingsWarningPopOverNotUsing: {
+    id: 'assets.warning.notUsing',
+    defaultMessage:
+      '!!!You are not using the recommended decimal place configuration for this native token.',
+    description: 'Asset settings recommended pop over content',
+  },
+});
+
+type Props = {
+  asset: AssetProps,
+  onCopyAssetItem?: Function,
+  highlightFingerprint?: boolean,
+  className?: string,
+};
+
+type State = {
+  itemCopied: ?string,
+};
+
+@observer
+export default class AssetContent extends Component<Props, State> {
```

Comment:
Let's write any "new" component with the latest React best practices (pure function + hooks)

Reply:
@DominikGuzei are we creating this kind of components already? It would be good to have some definitions, so we can keep it consistent.

Reply:
Done ✅